ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js

```diff
@@ -0,0 +1,214 @@
+                };
+
+                selectedAvailableType.visibleAttributes
+                    .filter((attribute) => !metacardDefinitions.isHiddenType(attribute))
+                    .filter((attribute) => !metacardDefinition[attribute].readOnly)
+                    .filter((attribute) => attribute !== "id")
+                    .forEach((attribute) => {
+                    if (metacardDefinition[attribute].multivalued) {
+                      propertyCollection[attribute] = [];
+                    } else if (metacardDefinitions.enums[attribute]) {
+                      propertyCollection[attribute] = metacardDefinitions.enums[attribute][0];
+                    } else {
+                      propertyCollection[attribute] = "";
+                    }
+                });
+
+                this.model.set('metacard', propertyCollection);
+                this.$el.addClass('is-building');
+
+            },
+            handleMetacard() {
+                const metacard = this.model.get('metacard');
+                this.builderProperties.show(PropertyCollectionView.generatePropertyCollectionView([metacard]));
+                this.builderProperties.currentView.$el.addClass("is-list");
+
+            },
+            onBeforeShow() {
+                    this.handleAvailableTypes();
+            },
+            edit() {
+                        this.$el.addClass('is-editing');
+                        this.builderProperties.currentView.turnOnEditing();
+                        this.builderProperties.currentView.focus();
+            },
+            cancel() {
+                        this.$el.removeClass('is-editing');
+                        this.builderProperties.currentView.revert();
+                        this.builderProperties.currentView.turnOffEditing();
+            },
+            save() {
+                        this.$el.removeClass('is-editing');
+
+                        const editedMetacard = this.builderProperties.currentView.toPropertyJSON([], []);
+
+                        const props = editedMetacard.properties;
+                        editedMetacard.properties = Object.keys(editedMetacard.properties)
+                            .filter((attributeName) => props[attributeName].length >= 1 && props[attributeName][0] !== "")
+                            .reduce((accummulator, currentValue) => _.extend(accummulator, { [currentValue]: props[currentValue]}), {});
+
+                        editedMetacard.properties['metacard-type'] = this.model.get('selectedAvailableType');
+
+                        $.ajax({
+                            type: 'POST',
+                            url: '/services/catalog/?transform=input-propertyjson',
+                            data: JSON.stringify(editedMetacard),
+                            contentType: 'application/json'
+                        }).then((response, status, xhr) => {
+                            this.options.handleNewMetacard(xhr.getResponseHeader('id'));
```

Comment:
✏️ Do we need to handle the possibility of a failure here?

Reply:
I added a check to make sure the id is returned. Did you have something more in mind?